inspirehep/utils/bibtex.py

```diff
@@ -0,0 +1,679 @@
+        if 'arxiv_eprints' in self.record:
+            for field in self.record['arxiv_eprints']:
+                if 'categories' in field:
+                    return field['categories'][0]
+        else:
+            return ''
+
+    def _get_series(self):
+        """Return series"""
+        series = ''
+        if 'book_series' in self.record:
+            for element in self.record['book_series']:
+                if 'value' in element:
+                    series = element['value']
+        return series
+
+    def _get_isbn(self):
+        """Return ISBN"""
+        isbn_list = []
+        result = []
+        if 'isbns' in self.record:
+            for isbn in self.record['isbns']:
+                isbn_list.append(isbn['value'])
+            if len(isbn_list) > 1:
+                for element in isbn_list:
+                    if isinstance(element, list):
+                        result.append(element[0])
+                    else:
+                        result.append(element)
+                return ', '.join(element for element in result)
+            else:
+                if any(isinstance(i, list) for i in isbn):
+                    nested_list = list(bibtex_booktitle.traverse(isbn))
+                    return nested_list[0]
+                else:
+                    return ''.join(result for result in isbn)
+        else:
+            return ''
+
+    def _get_pubnote(self):
+        """Return publication note"""
+        journal, volume, pages = ['', '', '']
+        if 'publication_info' in self.record:
+            pub_info = self.record['publication_info']
+            for field in pub_info:
+                if 'journal_title' in field:
+                    journal = field['journal_title']
+                if 'journal_volume' in field:
+                    volume = field['journal_volume']
+                if 'page_artid' in field:
+                    pages = field['page_artid']
+                    if pages:
+                        if isinstance(pages, list):
+                            for page in pages:
+                                dashpos = page.find('-')
+                                break
+                        else:
+                            dashpos = pages.find('-')
+                        if dashpos > -1:
+                            pages = pages.split('-')[0]
+                try:
+                    if journal and (volume != '' or pages != ''):
+                        recid = self.record['control_number', '']
+                        record = get_es_record_from_doc_type('literature',
+                                                             recid,
+                                                             'journals')
+                        coden = ','.join(
+                            [record['coden'][0], volume, pages])
+                        return coden
+                except:
+                    return ''
+        else:
+            return ''
```

Comment:
Ah no sorry. Now I see: I was misleaded by:
```python
coden = ','.join(['coden'][0], volume, pages])
```

that variable should actually be called pubnote (since it's the union of a journal title/coden, a volume and pages)